client/my-sites/checklist/wpcom-checklist/checklist-navigation/index.jsx

```diff
@@ -0,0 +1,104 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Children, Component } from 'react';
+import { connect } from 'react-redux';
+import { find, reduce } from 'lodash';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import ProgressBar from 'components/progress-bar';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+export class ChecklistNavigation extends Component {
+	static propTypes = {
+		siteSlug: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+		setStoredTask: PropTypes.func,
+		setNotification: PropTypes.func,
+		showNotification: PropTypes.bool,
+		canShowChecklist: PropTypes.bool,
+	};
+
+	handleClick = () => {
+		const childrenArray = Children.toArray( this.props.children );
+		const task = find( childrenArray, child => ! child.props.completed );
+
+		if ( task ) {
+			this.props.setStoredTask( task.key );
+		}
+		this.props.setNotification( false );
+		this.props.closePopover();
+		this.props.recordTracksEvent( 'calypso_inlinehelp_checklist_click' );
+	};
+
+	render() {
+		const { siteSlug, translate, showNotification, children, canShowChecklist } = this.props;
+
+		const buttonClasses = {
+			'has-notification': showNotification,
+			'checklist-navigation__count': true,
+		};
+		const childrenArray = Children.toArray( children );
```

Comment:
> From memory, that was problematic because the redux state is injected from an API response for the site checklist.

I think this is still true; using the selector, the checklist tasks come back as `undefined` when the app first loads, and it takes a few seconds to return them. They're not stored in state right away.

Reply:
Woooooo ... that's quite a nuance here. Thanks for the info. It could be something worth fixing, but it is totally not within the scope of this PR. So let's keep it this way :)